types/bull-arena/index.d.ts

```diff
@@ -1,56 +1,80 @@
-// Type definitions for bull-arena 2.6
+// Type definitions for bull-arena 3.0
 // Project: https://github.com/bee-queue/arena/
 // Definitions by: Levi Bostian <https://github.com/levibostian>
+//                 Eli Skeggs <https://github.com/skeggse>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.8
 
-import { RequestHandler } from "express";
-import { ClientOpts } from "redis";
+import BeeQueue = require('bee-queue');
+import BullQueue = require('bull');
+import { RequestHandler } from 'express';
+import { ClientOpts } from 'redis';
 
 declare function Arena(
-  options: BullArena.MiddlewareOptions,
-  listenOptions?: BullArena.MiddlewareListenOptions
+    options: BullArena.MiddlewareBeeOptions | BullArena.MiddlewareBullOptions | BullArena.MiddlewareOptions,
```

Comment:
Not super happy with how this ends up working - it almost feels like there should be a way to do this with type parameters. I'm coming from more of a flow background so maybe there's a way in typescript I haven't come across.

Reply:
Gotcha. I specifically wanted to enforce inclusion of the appropriate constructor when a given `ConfigQueue` had the corresponding `type`.